### PR TITLE
Docs: example for getSelectedBlock

### PR DESCRIPTION
**Pull request by troychaplin**
## What?
Adding an example for getSelectedBlock from the block-editor package

## Why?
Because examples are needed as per #42125 

## How?
Function had no example before

### CODE REVIEW

File: docs/reference-guides/data/data-core-block-editor.md
Comment (ryanwelcher) [inline docs/reference-guides/data/data-core-block-editor.md]: I'm not sure we need this part it doesn't add much to the example and might be taken as being needed to use it.

**Review by ryanwelcher — CHANGES_REQUESTED** [review]
Overall, looks great. Just one note around using the `subscribe`

File: docs/reference-guides/data/data-core-block-editor.md
Comment (troychaplin) [reply docs/reference-guides/data/data-core-block-editor.md]: Without a listener the function wouldn't log anything to the console, the subscribe would be needed here.

Comment (ryanwelcher) [reply docs/reference-guides/data/data-core-block-editor.md]: Right, but this could also be assigned to a button onClick or run in a useEffect for example.

Looking at some other [examples](https://developer.wordpress.org/block-editor/reference-guides/data/data-core/#getposttypes), we don't always show how the implementation should be used - just how to implement the item in question.

That being said, this is a strong opinion loosely held so I'll leave it up to you :) If you decide to keep it can we change the `subscribe` syntax to use imports instead of accessing the global?

Great work and keep them coming!

**Review by ryanwelcher — APPROVED** [review]
Approved with a note

File: docs/reference-guides/data/data-core-block-editor.md
Comment (troychaplin) [reply docs/reference-guides/data/data-core-block-editor.md]: If we remove subscribe is there a preferred way to add a note relating to a couple of use cases. I never thought about using it in an onClick, might be good to show both options in the docs.

Thoughts?

Comment (ryanwelcher) [reply docs/reference-guides/data/data-core-block-editor.md]: You could add it commented out similar to how it's [shown here](https://github.com/WordPress/wporg-developer-blog/issues/68)

Comment (troychaplin) [reply docs/reference-guides/data/data-core-block-editor.md]: > You could add it commented out similar to how it's [shown here](https://github.com/WordPress/wporg-developer-blog/issues/68)

That the correct link? Seems unrelated

Comment (ryanwelcher) [reply docs/reference-guides/data/data-core-block-editor.md]: 🤦 this is the correct link https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#useentityrecord

Comment (troychaplin) [reply docs/reference-guides/data/data-core-block-editor.md]: Thanks, added a couple of examples of how the function can be used

**Comment by troychaplin** [comment]
> Approved with a note

I know you approved but I made a followup change so if you could just take a quick peak before I merge that’d be great

**Review by ryanwelcher — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**
